### PR TITLE
Change admin website to Laravel with working user registration and classes crud

### PR DESCRIPTION
**Pull request by Daedalus11069**
In short: This PR adds the admin boilerplate and CRUD for the character classes to the admin website.

Long explanation: This PR makes use of the Laravel framework, currently at version 6.1.0 in this PR, to facilitate the admin website's update to a working MVC structure.  As of this PR, only the basic user registration and login, as well as CRUD for character classes is working.

There's a lot of files to review, but the important directories are:

- `app/Http/Controllers/`
- `config`
- `database/migrations/`
- `routes`
- `resources/views/classes/`

Here's a few important files:

- `app/Http/Middleware/Admin.php` -> Checks if the user at the supplied trust level is authorized to view the resource.
- `app/Http/Middleware/AdminMenuGenerator.php` -> Generates the admin menu.  Not large at this moment, but will be later.

Everything else is boilerplate or required files for Laravel to work.

This website will not work out of the box, not without a little bit of CLI; but that can be automated later for the build process of downloads/etc.

To get the website up and running, these are the commands that need to be ran, as well as their descriptions, and their requirements.  **All commands are ran from the admin directory**:

1. `composer install` -> Installs the php dependencies. Requires [Composer](https://getcomposer.org/) to be installed on the development system.  The `vendor` directory needs to be uploaded to the host system.
2. `npm install` -> Installs the frontend dependencies for building the frontend js and css files. Requires [NodeJS](https://nodejs.org/en/) to be installed on the development system.  I'd suggest going with 12.x LTS.
3. `npm run dev` -> Builds the frontend js and css files and sends those files to the public directory. Neither Nodejs nor `node_modules` needs to be on the host system. Requires the above(NodeJs), `dev` can be replaced with `prod` as required.
4. `cp ./.env.example ./.env` -> This file needs to be edited with the database details required by the host system to access the database, of course.  File config should be self explanatory, but `DB_HOST2` and all db-related configurations that follow are for the `rift_core` db.
5. `./artisan key:generate` -> Assuming that the artisan file is executable(otherwise it needs to be ran via php), this generates a unique key to the website that is used for password hashing or any encryption.  **This can be ran on the dev system and the modified `.env` file uploaded to the host.**
6. `./artisan migrate` Assuming that the artisan file is executable(otherwise it needs to be ran via php), this runs the migrations that add required tables to the database.  Of note, this should be ran last and only once the environment file is configured.  **This needs to be ran on the host or the changed database sql needs to be updated on the host.**

Once those commands are ran, and the environment file copied and configured, the website should be usable.  The host system will of course require apache/nginx with a virtual host or etc(I don't know what it's called for nginx) to work.

Lastly for install/usage, the virtual host or etc should point to the `public` directory.  The `.env` file can be symlinked from elsewhere, and for security, it is likely better that it is.

Of note, I got wired in while coding and didn't notice the update to the contribution guidelines.  I will be replacing all spaces with appropriately-sized tabs in the next PR.

### CODE REVIEW

**Comment by rezalas** [comment]
Question, does this also address issue numbers #12 #13 and #14 ? Just asking to know if we should assoc and close those issues too.

**Comment by Daedalus11069** [comment]
@rezalas All of those issues appear to be addressed by the PR:

- #12 I believe I have this one managed; I made the player directory a [configurable environment option](https://github.com/Daedalus11069/riftshadow/blob/admin-classes-crud/admin/.env.example#L9) for the `.env` file, and as to the strings, those can be replaced at a later time with language-dependent strings.
- #13 This is taken care of just by how Laravel works; that data is included in the `.env` file and then used throughout the laravel codebase very easily.  As an example, you can see [this PlayerFile model file](https://github.com/Daedalus11069/riftshadow/blob/admin-classes-crud/admin/app/Models/PlayerFile.php); all DB hits use the default connection type specified in the `.env` file, and can use [another one(ClassTable model file)](https://github.com/Daedalus11069/riftshadow/blob/admin-classes-crud/admin/app/Models/ClassTable.php) if specified.
- #14 This is taken care of using the facilities provided by Laravel; all input is escaped with their ORM.  Example of that [here, in the Classes store method](https://github.com/Daedalus11069/riftshadow/blob/admin-classes-crud/admin/app/Http/Controllers/ClassesController.php#L47).

So in short, I believe we should close those.  But as a note, in case it matters: I moved the old admin interface to `admin_old` for referencing as I built the new one.  So as long as access to that directory is disallowed, this should take care of #12, #13, and #14.
